src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.Searcher.cs

```diff
@@ -121,33 +127,48 @@ private class Searcher
                 }
             }
 
-#pragma warning disable CS0618 // MatchKind is obsolete
             private void ReportMatchResult(Project project, INavigateToSearchResult result)
             {
+                var builder = ArrayBuilder<Span>.GetInstance();
+                foreach (var s in result.NameMatchSpans)
+                {
+                    builder.Add(s.ToSpan());
+                }
+
+                var matchedSpans = builder.ToImmutableAndFree();
+
+                var patternMatch = new PatternMatch(GetPatternMatchKind(result.MatchKind), 
+                    punctuationStripped: true, result.IsCaseSensitive, matchedSpans);
+
                 var navigateToItem = new NavigateToItem(
                     result.Name,
                     result.Kind,
                     GetNavigateToLanguage(project.Language),
                     result.SecondarySort,
                     result,
-                    GetMatchKind(result.MatchKind),
-                    result.IsCaseSensitive,
+                    patternMatch,
                     _displayFactory);
                 _callback.AddItem(navigateToItem);
             }
 
-            private MatchKind GetMatchKind(NavigateToMatchKind matchKind)
+            private PatternMatchKind GetPatternMatchKind(NavigateToMatchKind matchKind)
             {
                 switch (matchKind)
                 {
-                    case NavigateToMatchKind.Exact: return MatchKind.Exact;
-                    case NavigateToMatchKind.Prefix: return MatchKind.Prefix;
-                    case NavigateToMatchKind.Substring: return MatchKind.Substring;
-                    case NavigateToMatchKind.Regular: return MatchKind.Regular;
-                    default: return MatchKind.None;
+                    case NavigateToMatchKind.Exact: return PatternMatchKind.Exact;
+                    case NavigateToMatchKind.Prefix: return PatternMatchKind.Prefix;
+                    case NavigateToMatchKind.Substring: return PatternMatchKind.Substring;
+                    case NavigateToMatchKind.Regular: return PatternMatchKind.Fuzzy;
+                    case NavigateToMatchKind.None: return PatternMatchKind.Fuzzy;
```

Comment:
I can't find in our code where we set NavigateToMatchKind.None. is this no longer used?